region/src/main/java/org/motechproject/nms/region/service/impl/DistrictServiceImpl.java

```diff
@@ -21,14 +25,19 @@
 
     private DistrictDataService districtDataService;
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(DistrictServiceImpl.class);
+
     @Autowired
     public DistrictServiceImpl(DistrictDataService districtDataService) {
         this.districtDataService = districtDataService;
     }
 
     @Override
+    @Cacheable("district-language")
```

Comment:
I think we can combine these per-method caches into a single cache using a custom key:

http://stackoverflow.com/a/14077831/379354